c7n/utils.py

```diff
@@ -610,6 +610,23 @@ def get_proxy_url(url):
         'all'
     ]
 
+    # Remove username:password if present in netloc.
+    target = re.sub(r'^.*@', '', url_parts['netloc'])
```

Comment:
most of this logic can be omitted

```python
>>> from urllib.parse import urlparse
>>> urlparse('https://user:pass@example.com:3030').hostname
"example.com"
```

Reply:
The logic just below this needs to stay in place to handle `no_proxy` definitions that contain ports, but yeah :614 could be re-written to use urllib instead of using an re. How would you like to see it handled? Currently this file already imports parse as urlparse, so your suggestion would have to be updated to:
```python
+>>> from urllib import parse as urlparse   # the current import in utils.py
+>>> urlparse.urlparse('https://user:pass@example.com:3030').hostname
'example.com'
```
Good with the `urlparse.urlparse`?

Reply:
Scratch that, updated `parse_url_config` to pass hostname and port. Let me know if there's anything else.

Reply:
could you clarify why your doing the extra handling on ports? no proxies are commonly against the host/domain non inclusive of port. are you saying your no proxy values contains port numbers?

Reply:
It looks like this is the implementation used in `request` 

"The no_proxy environment variable can be used to specify hosts which shouldn’t be reached via proxy; if set, it should be a comma-separated list of hostname suffixes, optionally with :port appended, for example cern.ch,ncsa.uiuc.edu,some.host:8080."

Reply:
The `proxy_bypass()` doesn't work properly if the port is not passed, but it is defined in the `no_proxy` env var:

```bash
export no_proxy=example.com:80
```
```python
+>>> from urllib.request import getproxies, proxy_bypass
+>>> proxy_bypass('example.com')
False
+>>> proxy_bypass('example.com:80')
True
```

---

```bash
export no_proxy=example.com
```
```python
+>>> from urllib.request import getproxies, proxy_bypass
+>>> proxy_bypass('example.com')
True
+>>> proxy_bypass('example.com:80')
True
```
These proxy env vars are challenging because they get implemented differently everywhere. This is needed for our particular use case and should not impact others.

Reply:
ic.. i don't think i've ever seen a port specification in a no_proxy env var. seems pretty rare.